src/app/calculator/motors/motor-performance/motor-performance-chart/motor-performance-chart.component.ts

```diff
@@ -0,0 +1,321 @@
+    if (expandedChart) {
+      if (this.expanded) {
+        this.currentChartId = this.expandedChartId;
+      }
+      else {
+        this.currentChartId = this.tabPanelChartId;
+      }
+      this.initRenderChart();
+    }
+  }
+
+  initRenderChart() {
+    Plotly.purge(this.currentChartId);
+    this.initChartSetup();
+    this.drawTraces();
+
+    let chartLayout = JSON.parse(JSON.stringify(this.performanceChart.layout));
+    Plotly.newPlot(this.currentChartId, this.performanceChart.data, chartLayout, this.performanceChart.config)
+      .then(chart => {
+        chart.on('plotly_click', chartData => {
+          this.addSelectedPointTraces(chartData);
+        });
+      });
+    this.save();
+  }
+
+  initChartSetup() {
+    this.graphColors = graphColors;
+    this.performanceChart = this.performanceChartService.performanceChart.getValue();
+    this.selectedDataPoints = this.performanceChartService.selectedDataPoints.getValue();
+  }
+
+  drawTraces() {
+    let allLineData: Array<TraceCoordinates> = this.performanceChartService.buildLineData(this.performanceForm, this.settings);
+    allLineData.forEach((data, index) => {
+      let trace = this.performanceChartService.getEmptyTrace();
+      trace.x = data.x;
+      trace.y = data.y;
+      trace.name = this.traceNames[index];
+      trace.hovertemplate = '%{y:.2r}%';
+      trace.line.color = this.graphColors[(index) % this.graphColors.length]
+      this.performanceChart.data[index] = trace;
+    });
+  }
+
+  addSelectedPointTraces(graphData) {
+    let currentPointIndex: number = graphData.points[0].pointIndex;
+    let selectedMotorPoint: MotorPoint = this.getMotorPoint(currentPointIndex, graphData.points[0]);
+    let currentColor = this.graphColors[(this.performanceChart.data.length - 1) % this.graphColors.length]
+    this.performanceChart.data.forEach((trace, i) => {
+      // Create points for 3 default line traces
+      if (i < 3) {
+        let point: MotorPoint = {
+          pointColor: currentColor,
+          current: selectedMotorPoint.current,
+          power: selectedMotorPoint.power,
+          efficiency: selectedMotorPoint.efficiency,
+          shaftLoad: selectedMotorPoint.shaftLoad,
+          x: Number(trace.x[currentPointIndex]),
+          y: Number(trace.y[currentPointIndex])
+        };
+        let pointTrace = this.performanceChartService.getTraceDataFromPoint(point);
+        let yMeasureTitle = this.traceNames[i];
+        let hoverTemplate = 'Motor Shaft Load' + ': %{x}%<br>' + yMeasureTitle + ': %{y:.2r}%';
+        pointTrace.hovertemplate = hoverTemplate;
+        Plotly.addTraces(this.currentChartId, pointTrace);
+      }
+    });
+    selectedMotorPoint.pointColor = currentColor;
+    this.selectedDataPoints.push(selectedMotorPoint);
+    this.cd.detectChanges();
+    this.save();
+  }
+
+  checkCurveChanged() {
+    if (this.tempMotorPower !== this.performanceForm.controls.horsePower.value) {
+      this.curveChanged = true;
```

Comment:
curveChanged should just be a local variable to this function. doesn't look like it's used anywhere else